pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -322,6 +328,16 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 		} else {
 			require.NotEqualValues(t, srcRes, destRes)
 		}
+
+		// Sanity: Execute the same query as prepared statement inside the reader
```

Comment:
to confirm, were these unit tests with prepared stmts failing before your patch to the source code?

Reply:
I didn't see any issues on the prepared statements, but opening new connections could fail intermittently.
<!-- Reviewable comment -OCJVU7N5_zkt0EhDOJb:b3grkjp -->
<!-- Sent from Reviewable.io -->